Replace akka-http-core-experimental with akka-http-core

tkqubo opened the pull request:
This PR will update akka libraries, to use akka-pusher with akka-http.
⚠️ after this change, spray can no longer be supported, as it doesn't run with akka-http 2.4.x

dtaniwaki commented:
@tkqubo Thank you for the PR! Could you remove unnecessary version change such as Scalia version and squash the commits?
I will do it outside this PR if necessary.

tkqubo committed:
chore: akka-http 2.4.9

tkqubo commented:
OK I removed unnecessary lines and tidied up commits :-)

And one thing I want to inform you that since Akka 2.4.0 dropped the support of Scala 2.10.x, I removed relevant lines.

Commit:
Enables to use named dispatcher.

tkqubo committed:
Merge pull request #10 from jkugiya/fix/use-execution-context-from-config

Enables to use named dispatcher.

dtaniwaki commented on src/main/scala/com/github/dtaniwaki/akka_pusher/PusherActor.scala:
```diff
@@ -23,7 +23,16 @@ class PusherActor(
   }
 
   implicit val system = context.system
-  implicit val ec: ExecutionContext = system.dispatcher
```
Is this change necessary to update Akka version?

dtaniwaki commented on src/test/resources/application.conf:
```diff
@@ -7,4 +7,14 @@ pusher {
   secret=${?PUSHER_API_SECRET}
   batchTrigger = false
   batchInterval = 1000
+  dispatcher {
```
ditto

dtaniwaki reviewed (changes requested):
@tkqubo Please check the comments and update the branch.

dtaniwaki commented on .travis.yml:
```diff
@@ -6,15 +6,13 @@ cache:
 script:
-  - "[ \"$TRAVIS_SCALA_VERSION\" != \"2.10.6\" ] || sbt ++$TRAVIS_SCALA_VERSION clean test"
-  - "[ \"$TRAVIS_SCALA_VERSION\" == \"2.10.6\" ] || sbt ++$TRAVIS_SCALA_VERSION clean coverage test"
+  - "sbt ++$TRAVIS_SCALA_VERSION clean coverage test"
```
Should be `clean test` if this change is for Scala 2.11.x?

Commit:
Remove dependency to system.dispatcher

tkqubo committed:
Merge pull request #11 from jkugiya/chore/akka-http

Remove dependency to system.dispatcher

dtaniwaki commented:
It seems your repository includes a lot of unnecessary changes. I will do the same change by myself. Thanks for the reporting!

tkqubo commented:
Sorry for not responding long time.  I will leave this to you.  Thanks!
